source/common/config/subscription_factory_impl.cc

```diff
@@ -112,6 +113,27 @@ SubscriptionPtr SubscriptionFactoryImpl::collectionSubscriptionFromUrl(
     return std::make_unique<Config::FilesystemCollectionSubscriptionImpl>(
         dispatcher_, path, callbacks, resource_decoder, stats, validation_visitor_, api_);
   }
+  case xds::core::v3::ResourceLocator::XDSTP: {
+    if (resource_type != collection_locator.resource_type()) {
+      throw EnvoyException(
+          fmt::format("xdstp:// type does not match {} in {}", resource_type,
+                      Config::XdsResourceIdentifier::encodeUrl(collection_locator)));
+    }
+    const envoy::config::core::v3::ApiConfigSource& api_config_source = config.api_config_source();
+    Utility::checkApiConfigSourceSubscriptionBackingCluster(cm_.primaryClusters(),
+                                                            api_config_source);
+
+    switch (api_config_source.api_type()) {
+    case envoy::config::core::v3::ApiConfigSource::AGGREGATED_DELTA_GRPC: {
+      return std::make_unique<GrpcCollectionSubscriptionImpl>(
+          collection_locator, cm_.adsMux(), callbacks, resource_decoder, stats, dispatcher_,
+          Utility::configSourceInitialFetchTimeout(config), false);
+    }
+    default:
+      throw EnvoyException(fmt::format("Unknown xdstp:// transport API type in {}",
+                                       api_config_source.DebugString()));
```

Comment:
Per the other site should this be NOT_IMPLEMENTED with a more clear comment that this is in progress? Or just a comment? Again mostly just trying to keep track of what is temporary and what is final.

Reply:
This one will be permanent, as we won't end up supporting every transport option. I'll add some comments. I've switched the `Unsupported collection resource locator` exception below to `NOT_IMPLEMENTED` though, as that one will eventually be added.